smpeg does not build with clang

not related though i found out why smpeg does not build with clang also, it seems to rely on gcc as the build horse and does not link by default to the clang rt builtins library leading to undefined symbols. I got around it temporarily by doing this ->

```
build() {
  [[ -d ${srcdir}/build-${MINGW_CHOST} ]] && rm -rf ${srcdir}/build-${MINGW_CHOST}
  mkdir -p "${srcdir}/build-${MINGW_CHOST}"
  cd "${srcdir}/build-${MINGW_CHOST}/" 
  case "${CARCH}" in
    i686)
      export LIBS+=" -lclang_rt.builtins-i386"
    ;;

    x86_64)
      export LIBS+=" -lclang_rt.builtins-x86_64"
    ;;
  esac
  CFLAGS+=" -Wno-error=narrowing"
  ../${_realname}/configure \
    --prefix=${MINGW_PREFIX} \
    --build=${MINGW_CHOST} \
    --host=${MINGW_CHOST} \
    --target=${MINGW_CHOST} \
    --mandir=${MINGW_PREFIX}/share/man \
    --enable-shared \
    --enable-static

  make
}
```

_Originally posted by @revelator in https://github.com/msys2/MINGW-packages/pull/9309#issuecomment-894637478_

Yes, the real problem is that it needs to be `autoreconf`-ed, because that's a matter of libtool not properly handling compiler-rt.  But, when I try to `autoreconf` it, I get link errors that `WinMain` isn't found...  This is from `smpeg2` but it's a similar situation with both.  344e671bfef59a55d7f17a58125416f48957d6b0
```
  libtool: link: g++ -march=x86-64 -mtune=generic -O2 -pipe -Wno-error=narrowing -ID:/a/_temp/msys/msys64/mingw64/include/SDL2 -Dmain=SDL_main -DTHREADED_AUDIO -DNDEBUG -Wall -I.. -DNOCONTROLS -I../smpeg2 -I../smpeg2/audio -I../smpeg2/video -pipe -Wl,--dynamicbase -Wl,--high-entropy-va -Wl,--nxcompat -Wl,--default-image-base-high -o .libs/plaympeg.exe plaympeg.o -mwindows  ./.libs/libsmpeg2.a -LD:/a/_temp/msys/msys64/mingw64/lib -lmingw32 -lSDL2main -lSDL2
  D:/a/_temp/msys/msys64/mingw64/lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys/msys64/mingw64/lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/lib/../lib/libmingw32.a(lib64_libmingw32_a-crt0_c.o): in function `main':
  C:/_/M/mingw-w64-crt-git/src/mingw-w64/mingw-w64-crt/crt/crt0_c.c:18: undefined reference to `WinMain'
  collect2.exe: error: ld returned 1 exit status
```

--- a/mingw-w64-smpeg/PKGBUILD
+++ b/mingw-w64-smpeg/PKGBUILD
@@ -4,7 +4,7 @@ _realname=smpeg
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 pkgver=0.4.5
-pkgrel=2
+pkgrel=3
 pkgdesc="SDL MPEG Player Library (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64')
@@ -17,13 +17,16 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
 depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs" "${MINGW_PACKAGE_PREFIX}-SDL")
 options=('staticlibs' 'strip')
 source=("${_realname}::svn://svn.icculus.org/smpeg/tags/release_${pkgver//./_}"
-        no-undefined.patch)
+        no-undefined.patch
+        clang-libtool-tweak.patch)
 sha256sums=('SKIP'
-            'd9c89f26abb9c5c6341876a613dac805e3c4ded590c99668104fcc453a4f412d')
+            'd9c89f26abb9c5c6341876a613dac805e3c4ded590c99668104fcc453a4f412d'
+            '0193f2f287b72372ec066f1057309e3ebf036c45473fecffea7cd7005ed23bc5')
 
 prepare() {
   cd "${srcdir}"/${_realname}
   patch -p1 -i ${srcdir}/no-undefined.patch
+  patch -p1 -i ${srcdir}/clang-libtool-tweak.patch
   ./autogen.sh
 }
 
@@ -33,6 +36,7 @@ build() {
   cd "${srcdir}/build-${MINGW_CHOST}/"
 
   CFLAGS+=" -Wno-error=narrowing"
+  lt_cv_deplibs_check_method=pass_all \
   ../${_realname}/configure \
     --prefix=${MINGW_PREFIX} \
     --build=${MINGW_CHOST} \

--- a/mingw-w64-smpeg/clang-libtool-tweak.patch
+++ b/mingw-w64-smpeg/clang-libtool-tweak.patch
@@ -1,0 +1,11 @@
+--- smpeg/acinclude/libtool.m4~	2021-08-08 11:05:04.022380400 -0700
++++ smpeg/acinclude/libtool.m4	2021-08-08 14:24:36.675795800 -0700
+@@ -6419,7 +6419,7 @@
+   for p in `eval "$output_verbose_link_cmd"`; do
+     case $p in
+ 
+-    -L* | -R* | -l*)
++    -L* | -R* | -l* | */libclang_rt.*.a)
+        # Some compilers place space between "-{L,R}" and the path.
+        # Remove the space.
+        if test $p = "-L" ||

--- a/mingw-w64-smpeg2/PKGBUILD
+++ b/mingw-w64-smpeg2/PKGBUILD
@@ -4,7 +4,7 @@ _realname=smpeg2
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 pkgver=2.0.0
-pkgrel=5
+pkgrel=6
 pkgdesc="SDL2 MPEG Player Library (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64')
@@ -13,11 +13,14 @@ license=("LGPL")
 makedepends=("${MINGW_PACKAGE_PREFIX}-gcc" "subversion")
 depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs" "${MINGW_PACKAGE_PREFIX}-SDL2")
 options=('staticlibs' 'strip')
-source=("${_realname}::svn://svn.icculus.org/smpeg/tags/release_${pkgver//./_}")
-sha256sums=('SKIP')
+source=("${_realname}::svn://svn.icculus.org/smpeg/tags/release_${pkgver//./_}"
+        "clang-libtool-tweak.patch")
+sha256sums=('SKIP'
+            '0193f2f287b72372ec066f1057309e3ebf036c45473fecffea7cd7005ed23bc5')
 
 prepare() {
   cd "${srcdir}"/${_realname}
+  patch -p1 -i "${srcdir}/clang-libtool-tweak.patch"
   ./autogen.sh
 }
 
@@ -27,6 +30,7 @@ build() {
   cd "${srcdir}/build-${MINGW_CHOST}/"
 
   CFLAGS+=" -Wno-error=narrowing"
+  lt_cv_deplibs_check_method=pass_all \
   ../${_realname}/configure \
     --prefix=${MINGW_PREFIX} \
     --build=${MINGW_CHOST} \

--- a/mingw-w64-smpeg2/clang-libtool-tweak.patch
+++ b/mingw-w64-smpeg2/clang-libtool-tweak.patch
@@ -1,0 +1,11 @@
+--- smpeg/acinclude/libtool.m4~	2021-08-08 11:05:04.022380400 -0700
++++ smpeg/acinclude/libtool.m4	2021-08-08 14:24:36.675795800 -0700
+@@ -6419,7 +6419,7 @@
+   for p in `eval "$output_verbose_link_cmd"`; do
+     case $p in
+ 
+-    -L* | -R* | -l*)
++    -L* | -R* | -l* | */libclang_rt.*.a)
+        # Some compilers place space between "-{L,R}" and the path.
+        # Remove the space.
+        if test $p = "-L" ||

